src/redo.erl

```diff
@@ -273,11 +273,23 @@ handle_info({tcp_closed, Sock}, #state{sock=Sock}=State) ->
             {noreply, State1#state{sock=undefined}, 1000}
     end;
 
+handle_info({tcp_error, Sock, _Reason}, #state{sock=Sock, reconnect=true}=State) ->
+    State1 = close_connection(State),
+
```

Comment:
add some logging in case of an error. Otherwise it will be reconnecting forever and we wouldn't have any clue - what is going on.